thinkhazard_processing/processing.py

```diff
@@ -0,0 +1,145 @@
+# -*- coding: utf-8 -*-
+
+import transaction
+import datetime
+import numpy
+import rasterio
+import pyproj
+from rasterio import features
+from shapely.ops import transform
+from functools import partial
+from geoalchemy2.shape import to_shape
+
+from thinkhazard_common.models import (
+    DBSession,
+    AdministrativeDivision,
+    HazardLevel,
+    )
+from .models import (
+    HazardSet,
+    Layer,
+    Output,
+    )
+from . import settings
+
+
+class ProcessException(Exception):
+    def __init__(self, message):
+        self.message = message
+
+
+def process_pendings():
+    ids = DBSession.query(HazardSet.id) \
+        .filter(HazardSet.complete.is_(True)) \
+        .filter(HazardSet.processed.is_(False))
+    for id in ids:
+        process_hazardset(id)
+
+
+def process_all():
+    ids = DBSession.query(HazardSet.id) \
+        .filter(HazardSet.complete.is_(True))
+    for id in ids:
+        process_hazardset(id, force=True)
```

Comment:
I think this could be simplified to something like:
```python
def process_hazardsets(all=False):
    ids = DBSession.query(HazardSet.id) \
        .filter(HazardSet.complete.is_(True))
    if not all:
        ids.filter(HazardSet.processed.is_(False))
    for id in ids:
        process_hazardset(id, force=all)
```